test/cpp-raw/interface_params.cpp

```diff
@@ -7,13 +7,13 @@
 
 #include <intercom.hpp>
 
-class CppImplementation : public ISharedInterface
+class CppImplementation : public ISharedInterface_Automation
```

Comment:
I think we should always, if automation is specified, use the base name for COM automation. It will be annoying if one needs to include the _Automation word after every interface when it is used from, a scripting language. From C++ it is much easier, and idiomatic, to just use "type_lit::raw::*", and one can use "using" statement reduce the boilerplate.

Note: I could not find an IDL [attrribute](https://docs.microsoft.com/en-us/cpp/windows/idl-attributes) that would have allowed specifying a custom name for the interface for automation purposes.

Reply:
In most use cases, one would not include both type systems in scripting language IDLs. In this case the Automation interface uses the default `ISharedInterface` naming. C++ wrappers in this case use `ISharedInterface` for the raw interface. The interface names become `ISharedInterface_Automation`/`ISharedInterface_Raw` only when the IDL/C++ wrappers contain both of these.

Reply:
This is mostly a testing feature - as for the `cpp-raw` tests we will want to invoke both the `_Automation` interfaces and the `_Raw` interfaces.